src/snowflake/cli/api/project/util.py

```diff
@@ -12,8 +12,11 @@
 SINGLE_QUOTED_STRING_LITERAL_REGEX = r"'((?:\\.|''|[^'\n])+?)'"
 
 # See https://docs.snowflake.com/en/sql-reference/identifiers-syntax for identifier syntax
-UNQUOTED_IDENTIFIER_REGEX = r"(^[a-zA-Z_])([a-zA-Z0-9_$]{0,254})"
+UNQUOTED_IDENTIFIER_REGEX = r"([a-zA-Z_])([a-zA-Z0-9_$]{0,254})"
 QUOTED_IDENTIFIER_REGEX = r'"((""|[^"]){0,255})"'
```

Comment:
As far as I can see, all uses of UNQUOTED_IDENTIFIER_REGEX are used with re.fullmatch (as they should), so I believe this start-of-string character is unnecessary